### ISSUE TITLE
Suggestion: advertise --

### ISSUE THREAD

**Issue by ChillerDragon** [koalaman/shellcheck#2121]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Advertise the double dash when an argument is passed as string containing leading dashes.
Sometimes one wants to ``printf`` something with leading dashes or ``grep`` for something beginning with --.
If quotes are used this can be seen as signal this is not intended to be an option but a value.

If the -- is not used printf for example breaks:
```
-bash: printf: --: invalid option
printf: usage: printf [-v var] format [arguments]
```

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

printf "--- foo ---"
```

#### Here's what I wanted or expected to see:

```
In test.sh line 3:
    printf "--- foo ---"
            ^-- SCXXXX: Consider using double dashes in front of arguments with leading dashes

Did you mean:
    printf -- "--- foo ---"
```

Somewhat related to #314

**Comment by guillaume-d** [comment]
Also related to #2157 (kind of a special case of it), but here the case is more clear-cut (literal string beginning with dashes rather than an inscrutable quoted variable expansion), maybe allowing a better diagnostic message and much more confidence.
